xbmc/platform/android/filesystem/AndroidAppFile.cpp

```diff
@@ -139,17 +139,18 @@ unsigned int CFileAndroidApp::ReadIcon(unsigned char** lpBuf, unsigned int* widt
     return 0;
   }
 
-  *width = info.width;
-  *height = info.height;
-
-  int imgsize = *width * *height * 4;
-  *lpBuf = new unsigned char[imgsize];
+  int imgsize = info.width * info.height * 4;
+  auto buffer = std::make_unique<unsigned char[]>(imgsize);
 
   AndroidBitmap_lockPixels(env, bmp.get_raw(), &bitmapBuf);
   if (bitmapBuf)
   {
     memcpy(*lpBuf, bitmapBuf, imgsize);
     AndroidBitmap_unlockPixels(env, bmp.get_raw());
+
+    *width = info.width;
+    *height = info.height;
+    *lpBuf = buffer.release();
```

Comment:
Need a bit above, on `memcpy(*lpBuf, bitmapBuf, imgsize);` not available then.

Reply:
Fixed.

Reply:
Maybe as:
```cpp
  AndroidBitmap_lockPixels(env, bmp.get_raw(), &bitmapBuf);
  if (bitmapBuf)
  {
    int imgsize = info.width * info.height * 4;
    *lpBuf = new unsigned char[imgsize];
    *width = info.width;
    *height = info.height;

    memcpy(*lpBuf, bitmapBuf, imgsize);
    AndroidBitmap_unlockPixels(env, bmp.get_raw());

    return imgsize;
  }
```

I'm not 100% sure, but as the memory seems only used inside this if a alloc seems not needed.

Reply:
That's definitely better :+1: